stl/inc/xstring

```diff
@@ -2242,10 +2242,10 @@ public:
         return _Result;
     }
 
-    _NODISCARD static _CONSTEXPR20 bool _Large_string_engaged(size_type _Strcap) noexcept {
-        // decides whether string is in large mode; has no implication on largeness of string's size
-        _STL_INTERNAL_CHECK(_Strcap >= _SMALL_STRING_CAPACITY);

```

Comment:
It was this check(added in this pr) that were causing lots of test failures, as
1. `_Myres` is initialized to `0` on construction.
2. In several places(like [`_Construct`](https://github.com/microsoft/STL/blob/2261f7edb760eb3fe0726187c818b796dc7ea798/stl/inc/xstring#L2610-L2612)), `_Large_string_engaged` is called early before assignment of `_Myres`.

After making `_Myres` initialized to `_SMALL_STRING_CAPACITY`, I made some checks passed, however there were some errors in clang checks due to `constexpr evaluation hit maximum step limit`.

Here I decide to remove this check to pass those clang tests. However, I'm uncertain wether this check is helpful. **Does the check make sense? Should I add it back?**

And I'm leaving `_Myres` initialized to `_SMALL_STRING_CAPACITY`, as `_String_val` begins with SSO mode. This change will make some assignments redundant: (Candidates: [(1)](https://github.com/microsoft/STL/blob/2261f7edb760eb3fe0726187c818b796dc7ea798/stl/inc/xstring#L2630) [(2)](https://github.com/microsoft/STL/blob/2261f7edb760eb3fe0726187c818b796dc7ea798/stl/inc/xstring#L2650) [(3)](https://github.com/microsoft/STL/blob/2261f7edb760eb3fe0726187c818b796dc7ea798/stl/inc/xstring#L2686); they are unchanged now)